tests/test_plugin_features.py

```diff
@@ -21,13 +28,143 @@ def cleanup_plugin_features():
     remove_all_plugin_features()
 
 
-def get_plugin_file(plugin_name="plugin_test_example.py"):
-    plugin_path = pathlib.Path(__file__).parent / "data" / plugin_name
-    return plugin_path
+def compute_single_plugin_feature(rtdc_ds):
+    circ_per_area = rtdc_ds["circ"] / rtdc_ds["area_um"]
+    return circ_per_area
+
+
+def compute_multiple_plugin_features(rtdc_ds):
+    circ_per_area = rtdc_ds["circ"] / rtdc_ds["area_um"]
+    circ_times_area = rtdc_ds["circ"] * rtdc_ds["area_um"]
+    return {"circ_per_area": circ_per_area, "circ_times_area": circ_times_area}
+
+
+def example_plugin_feature_info():
+    feature_info = {
+        "feature_name": "circ_per_area",
+        "method": compute_single_plugin_feature,
+        "req_config": [],
+        "req_features": ["circ", "area_um"],
+        "req_func": lambda x: True,
+        "priority": 1,
+    }
+    return feature_info
+
+
+def test_pf_attributes():
+    plugin_path = data_dir / "plugin_test_example.py"
+    plugin_list = dclab.load_plugin_feature(plugin_path)
+    pf1, pf2 = plugin_list
+
+    plugin_file_info = import_plugin_feature_script(plugin_path)
+
+    assert pf1.plugin_path == plugin_path
+    assert pf2.plugin_path == plugin_path
+    assert pf1.plugin_info == plugin_file_info
+    assert pf2.plugin_info == plugin_file_info
+    assert pf1.feature_label == plugin_file_info["feature labels"][0]
+    assert pf2.feature_label == plugin_file_info["feature labels"][1]
+    assert pf1.is_scalar
+    assert pf2.is_scalar
+    assert pf1.feature_name == plugin_file_info["feature names"][0]
+    assert pf2.feature_name == plugin_file_info["feature names"][1]
 
 
-def test_create_plugin():
-    plugin_path = get_plugin_file()
+def test_pf_bad_plugin_feature_name():
+    """Basic test of a bad name for PlugInFeature"""
+    plugin_path = ''
+    other_info = {}
+    feature_label = ""
+    is_scalar = True
+    feature_info = example_plugin_feature_info()
+    feature_info["feature_name"] = "Peter-Pan's Best Friend!"
+    with pytest.raises(ValueError):
+        PlugInFeature(feature_label, is_scalar, plugin_path,
+                      other_info, **feature_info)
+
+
+def test_pf_exists_in_hierarchy():
+    """Test for RTDCHierarchy"""
+    plugin_path = ''
+    other_info = {}
+    feature_label = ""
```

Comment:
Feature labels should never be empty strings (this should not work). They may be `None`, but then `dfn` will take care of it. Initially, this should be handled by `load_plugin_feature`.

Reply:
As discussed earlier, I have a check for this. IF the user for whatever reason leaves labels as empty strings, they will be set to None, and then the dfn file will handle it as usual.